api/v1/installation_types.go

```diff
@@ -926,6 +926,24 @@ type CNISpec struct {
 	// Calico Enterprise installation.
 	// +optional
 	IPAM *IPAMSpec `json:"ipam"`
+
+	// BinDir is the path to the CNI binaries directory.
+	// If you have changed the installation directory for CNI binaries in the container runtime configuration,
+	// please ensure that this field points to the same directory as specified in the container runtime settings.
+	// Default: /opt/cni/bin
+	// +optional
+	// +kubebuilder:default:="/opt/cni/bin"
+	// +kubebuilder:validation:Type=string
+	BinDir string `json:"binDir,omitempty"`
```

Comment:
Optional fields should be pointers in the API per conventions: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required


```suggestion
	BinDir *string `json:"binDir,omitempty"`
```

Reply:
Fixed in the following commit.

https://github.com/tigera/operator/pull/3882/commits/fa7184a673b869063c01fba32e5d42d2614ff532